src/server/game/AI/CoreAI/PetAI.cpp

```diff
@@ -113,19 +111,20 @@ bool PetAI::_canMeleeAttack() const
                     case IMP_FIREBOLT_RANK_9:
                     case WATER_ELEMENTAL_WATERBOLT_1:
                     case WATER_ELEMENTAL_WATERBOLT_2:
-                        canAttack = true;
+                        return true;
                         break;
                     default:
                         break;
                 }
             }
+            return false;
             break;
         }
         default:
             break;
     }
 
-    return canAttack;
+    return true;
```

Comment:
this should be false...

Reply:
No, this is correct, the default is "true". The function should only return false if either the Imp or the Water Elemental is used and autocast for the appropriate combat spell (Firebolt / Waterbolt) is disabled. Those two pets are special as they must not use melee attacks if autocast for their combat spells is disabled.

Reply:
oh ok, my bad. I confused with line 120.